File: YukariLauncher/src/main/java/net/kdt/pojavlaunch/plugins/FFmpegPlugin.java
package net.kdt.pojavlaunch.plugins;

import android.content.Context;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;

import com.arata.yukarilauncher.feature.log.Logging;

import java.io.File;

/**
 * FFmpegプラグインを検出して管理するクラス
 */
public class FFmpegPlugin {
    public static boolean isAvailable = false;
    public static String libraryPath;
    public static String executablePath;

    /**
     * FFmpegプラグインがインストールされているかどうかを検出します。
     * パッケージマネージャーからプラグイン情報を取得し、
     * ライブラリパスと実行可能パスを設定します。
     * @param context アプリケーションコンテキスト
     */
    public static void discover(Context context) {
        PackageManager manager = context.getPackageManager();
        try {
            PackageInfo ffmpegPluginInfo = manager.getPackageInfo("net.kdt.pojavlaunch.ffmpeg", PackageManager.GET_SHARED_LIBRARY_FILES);
            libraryPath = ffmpegPluginInfo.applicationInfo.nativeLibraryDir;
            File ffmpegExecutable = new File(libraryPath, "libffmpeg.so");
            executablePath = ffmpegExecutable.getAbsolutePath();
            // 古いプラグインバージョンにはまだ古い実行可能ファイルの場所がある
            isAvailable = ffmpegExecutable.exists();
        }catch (Exception e) {
            Logging.i("FFmpegPlugin", "Failed to discover plugin", e);
        }
    }
}

File: YukariLauncher/src/main/java/com/arata/yukarilauncher/plugins/FFmpegPlugin.kt
package com.arata.yukarilauncher.plugins

import android.content.Context
import com.arata.yukarilauncher.feature.log.Logging
import java.io.File

/** FFmpegプラグインを管理するオブジェクト。 */
object FFmpegPlugin {
    /** プラグインが利用可能か */
    var isAvailable = false
    /** ライブラリパス */
    var libraryPath: String? = null
    /** 実行可能ファイルパス */
    var executablePath: String? = null

    /** FFmpegプラグインを探索する。 @param context コンテキスト */
    fun discover(context: Context) {
        val manager = context.packageManager
        try {
            val ffmpegPluginInfo = manager.getPackageInfo("net.kdt.pojavlaunch.ffmpeg", 0)
            libraryPath = ffmpegPluginInfo.applicationInfo!!.nativeLibraryDir
            val ffmpegExecutable = File(libraryPath, "libffmpeg.so")
            executablePath = ffmpegExecutable.absolutePath
            isAvailable = ffmpegExecutable.exists()
        } catch (e: Exception) {
            Logging.i("FFmpegPlugin", "Failed to discover plugin", e)
        }
    }
}